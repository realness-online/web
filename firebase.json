{
  "storage": {
    "rules": "storage.rules"
  },
  "hosting": {
    "public": "dist",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ],
    "rewrites": [
      {
        "source": "**",
        "destination": "/index.html"
      }
    ],
    "headers": [
      {
        "source": "**",
        "headers": [
          {
            "key": "Content-Security-Policy",
            "value": "script-src 'unsafe-eval' 'unsafe-inline' 'self' https://www.google.com https://apis.google.com https://www.gstatic.com https://storage.googleapis.com/"
          }
        ]
      },
      {
        "source": "**",
        "headers": [
          {
            "key": "Cache-Control",
            "value": "max-age=31536000"
          }
        ]
      },
      {
        "source": "/service.worker.js",
        "headers": [
          {
            "key": "Cache-Control",
            "value": "no-cache"
          }
        ]
      }
    ]
  }
}
