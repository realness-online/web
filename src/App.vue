<style src="@/style/index.styl" lang="stylus"></style>
<template lang="html">
  <main id="realness">
    <h6>Version {{version}}</h6>
    <router-view></router-view>
    <developer-tools></developer-tools>
  </main>
</template>
<script>
  import developer_tools from '@/components/developer'
  import * as firebase from 'firebase/app'
  export default {
    watch: {
      '$route' (to, from) {
        sessionStorage.previous = from.path
      }
    },
    data() {
      return {
        version: process.env.VUE_APP_VERSION
      }
    },
    created() {
      console.log('initialize app')
      let config = {
        apiKey: process.env.VUE_APP_API_KEY,
        authDomain: process.env.VUE_APP_AUTH_DOMAIN,
        databaseUrl: process.env.VUE_APP_DATABASE_URL,
        projectId: process.env.VUE_APP_PROJECT_ID,
        storageBucket: process.env.VUE_APP_STORAGE_BUCKET,
        messagingSenderId: process.env.VUE_APP_MESSAGING_SENDER_ID
      }
      firebase.initializeApp(config)
    },
    components: {
      'developer-tools': developer_tools
    }
  }
</script>
<style lang="stylus">
  @require '/style/variables'
  #realness > h6
    position:fixed
    bottom: base-line
    left: base-line
</style>
