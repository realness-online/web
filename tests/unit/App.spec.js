import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import firebase from 'firebase/app'
import App from '@/App'
const fetch = require('jest-fetch-mock')
describe('@/App.vue', () => {
  let wrapper
  const node_env = process.env
  beforeEach(async () => {
    jest.resetModules()
    process.env = { ...node_env }
    wrapper = await shallowMount(App, {
      stubs: ['router-link', 'router-view']
    })
  })
  afterEach(() => {
    firebase.initializeApp.mockClear()
    wrapper.unmounted()
  })
  afterAll(() => {
    process.env = node_env
  })
  describe('Renders', () => {
    it('Layout of the application', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
    it('Initialises firebase', () => {
      expect(firebase.initializeApp).toHaveBeenCalled()
    })
    it('Initialises firebase in production', async () => {
      process.env.NODE_ENV = 'production'
      fetch.mockResponseOnce('{}')
      wrapper = await shallowMount(App, { stubs: ['router-link', 'router-view'] })
      expect(firebase.initializeApp).toHaveBeenCalled()
      expect(fetch.mock.calls.length).toBe(1)
      expect(fetch.mock.calls[0][0]).toBe('__/firebase/init.json')
    })
    it('Calls offline is app is initialized offline', async () => {
      jest.spyOn(window.navigator, 'onLine', 'get').mockReturnValueOnce(false)
      wrapper = await shallowMount(App, { stubs: ['router-link', 'router-view'] })
    })
  })
  describe('Routing', () => {
    it('Sets previously visited page in sessionStorage', async () => {
      const localVue = createLocalVue()
      localVue.use(VueRouter)
      const router = new VueRouter({
        routes: [
          { path: '/relations', name: 'relations' },
          { path: '/magic', name: 'magic' }
        ]
      })
      wrapper = shallowMount(App, { localVue, router })
      await router.push({ name: 'relations' })
      await router.push({ name: 'magic' })
      expect(sessionStorage.previous).toBe('/relations')
    })
  })
  describe('Methods', () => {
    describe('#onLine', () => {
      it('Turns the editable content back on', () => {
        const elements = [{ setAttribute: jest.fn() }]
        wrapper.vm.status = 'offline'
        jest.spyOn(document, 'querySelectorAll').mockReturnValueOnce(elements)
        wrapper.vm.online()
        expect(wrapper.vm.status).toBe(null)
        expect(elements[0].setAttribute).toBeCalled()
      })
    })
    describe('#offLine', () => {
      it('Turns the editable content back on', () => {
        const elements = [{ setAttribute: jest.fn() }]
        wrapper.vm.status = 'offline'
        jest.spyOn(document, 'querySelectorAll').mockReturnValueOnce(elements)
        wrapper.vm.offline()
        expect(wrapper.vm.status).toBe('offline')
        expect(elements[0].setAttribute).toBeCalled()
      })
    })
    describe('#sync_active', () => {
      it('Sets Status to active whebn syncing', () => {
        wrapper.vm.status = 'offline'
        wrapper.vm.sync_active(true)
        expect(wrapper.vm.status).toBe('working')
      })
      it('Sets Status to null when not syncing', () => {
        wrapper.vm.status = 'offline'
        wrapper.vm.sync_active(false)
        expect(wrapper.vm.status).toBe(null)
      })
    })
  })
})
