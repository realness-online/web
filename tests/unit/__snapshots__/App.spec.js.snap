// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/App.vue Renders Layout of the application 1`] = `
<main
  id="realness"
>
  <router-view-stub />
   
  <aside>
    <developer-tools-stub />
     
    <sync-stub
      config="[object Object]"
    />
  </aside>
</main>
`;
